src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java

```diff
@@ -107,8 +107,7 @@ public void testNumberFormatException() throws Exception {
     @Test
     public void testCustomTag() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(TempCheck.class);
-        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
-        verify(checkConfig, getPath("InputAbstractJavadocCustomTag.java"), expected);
+        execute(checkConfig, getPath("InputAbstractJavadocCustomTag.java"));
```

Comment:
let try https://github.com/checkstyle/checkstyle/issues/11446#issuecomment-1817540300

Reply:
It is leading to 

```
Violation lines for /Users/suniti/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocJavadocTagsWithoutArgs.java differ.
expected to be empty
but was: [16, 29, 35, 46, 68, 77]
```

Although in debugging, the expected is not resulting in empty value -
`
["14: Javadoc com...", "27: Javadoc com...", "33: Javadoc com...", "44: Javadoc com...", "66: Javadoc com...", +1 more]`

Reply:
If we try to add the word Config - 

```
/*
Config: com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest$TempCheck


*/
package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;

import java.io.Serializable;
```
it leads to 

`com.puppycrawl.tools.checkstyle.api.CheckstyleException: Config comment not specified properly in /Users/suniti/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocJavadocTagsWithoutArgs.java`

Reply:
Please try to put violations to `final String[] expected`

Reply:
We do not need word "config" 
Example https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethod1.java

Reply:
The input file `AbstractJavadocCustomTag` is working fine after the following changes 

```
% git diff master -- src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCustomTag.java
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCustomTag.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCustomTag.java
index ae99840d8..2122bf082 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCustomTag.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocCustomTag.java
@@ -1,8 +1,9 @@
-package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
-
 /*
- * Config: TempCheck
- */
+com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest$TempCheck
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
 
 public class InputAbstractJavadocCustomTag {
```

and the test method associated with it is also passing - 

```
checkstyle % git diff master -- src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java                          
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java
index e0ef35670..8e99bc463 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java

 public void testCustomTag() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(TempCheck.class);
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
-        verify(checkConfig, getPath("InputAbstractJavadocCustomTag.java"), expected);
+        verifyWithInlineConfigParser(getPath("InputAbstractJavadocCustomTag.java"), expected);
     }

```

Reply:
While doing the same changes in other input files and test methods whose expected violations are not empty, we are getting this for each one of the input files - 

```
Violation lines for /Users/suniti/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNumberFormatException.java differ.
expected to be empty
but was: [8]
```

Expected is always coming as empty, whereas debugging shows something different - 

`expected = {String[1]@2415}["8: Javadoc comm..."]`

Let's try to resolve the issue for this input class as it has only one violation and is easy to interrogate. Once we are done here, we can do the same for other input files.

```
checkstyle % git diff -- master src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNumberFormatException.java
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNumberFormatException.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNumberFormatException.java
index d51033e1e..87d35517b 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNumberFormatException.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNumberFormatException.java
@@ -1,8 +1,9 @@
-package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
-
 /*
- * Config: TempCheck
- */
+com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest$TempCheck
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
 
 /** <ul><li>a' {@link EntityEntry} (by way of {@link #;}</li></ul> */
 // violation 'Javadoc comment at column 52 has parse error.

```

Reply:
Please push changes to GitHub to let us see whole solution

Reply:
> Please try to put violations to `final String[] expected`

Yes, that's already there

Reply:
seems like we have found a way